lib/HLSL/DxilRootSignature.cpp

```diff
@@ -207,7 +207,7 @@ HRESULT SimpleSerializer::ReserveBlock(void **ppData, unsigned cbSize,
 
 Cleanup:
   if (FAILED(hr)) {
-    delete[] pClonedData;
+    delete[] (char*)pClonedData;
```

Comment:
according to [this](http://www.cplusplus.com/reference/new/operator%20delete[]/), `operator delete []` does take a `void*` 
pClonedData is also a void* as far as I can tell. I'm wondering why this cast is needed?

Reply:
The linked reference does indicate a void pointer, but I think that's to indicate that any pointer is valid. They don't really mean that a void pointer is a good idea in all cases. The canonical example is when a class has a destructor. If you delete it as a void, the destructor doesn't get called. [This](https://www.viva64.com/en/w/v772/) gives a pretty succinct summary and even cites the spec, which seems like a great way to add credibility to an argument. Especially when most people haven't paid up to get access to it. You could say it says anything! 

That said, we are using PODs here. So maybe the compiler is protesting too much. I don't see much point in the pointer ever being void. I'll try a few things and better nail down the specific warning and alternatives.